RecoMuon/TrackerSeedGenerator/plugins/MuonHLTSeedMVAClassifier.cc

```diff
@@ -0,0 +1,282 @@
+  const double baseScore_;
+
+  double getSeedMva(const pairSeedMvaEstimator& pairMvaEstimator,
+                    const TrajectorySeed& seed,
+                    const GlobalVector& global_p,
+                    const l1t::MuonBxCollection& l1Muons,
+                    const reco::RecoChargedCandidateCollection& l2Muons);
+};
+
+MuonHLTSeedMVAClassifier::MuonHLTSeedMVAClassifier(const edm::ParameterSet& iConfig)
+    : seedToken_(consumes<TrajectorySeedCollection>(iConfig.getParameter<edm::InputTag>("src"))),
+      l1MuonToken_(consumes<l1t::MuonBxCollection>(iConfig.getParameter<edm::InputTag>("L1Muon"))),
+      l2MuonToken_(consumes<reco::RecoChargedCandidateCollection>(iConfig.getParameter<edm::InputTag>("L2Muon"))),
+      trackerGeometryToken_(esConsumes<TrackerGeometry, TrackerDigiGeometryRecord>()),
+
+      rejectAll_(iConfig.getParameter<bool>("rejectAll")),
+      isFromL1_(iConfig.getParameter<bool>("isFromL1")),
+
+      mvaFileB_(iConfig.getParameter<edm::FileInPath>(isFromL1_ ? "mvaFileBL1" : "mvaFileBL2")),
+      mvaFileE_(iConfig.getParameter<edm::FileInPath>(isFromL1_ ? "mvaFileEL1" : "mvaFileEL2")),
+
+      mvaScaleMeanB_(iConfig.getParameter<std::vector<double>>(isFromL1_ ? "mvaScaleMeanBL1" : "mvaScaleMeanBL2")),
+      mvaScaleStdB_(iConfig.getParameter<std::vector<double>>(isFromL1_ ? "mvaScaleStdBL1" : "mvaScaleStdBL2")),
+      mvaScaleMeanE_(iConfig.getParameter<std::vector<double>>(isFromL1_ ? "mvaScaleMeanEL1" : "mvaScaleMeanEL2")),
+      mvaScaleStdE_(iConfig.getParameter<std::vector<double>>(isFromL1_ ? "mvaScaleStdEL1" : "mvaScaleStdEL2")),
+
+      doSort_(iConfig.getParameter<bool>("doSort")),
+      nSeedsMaxB_(iConfig.getParameter<int>("nSeedsMaxB")),
+      nSeedsMaxE_(iConfig.getParameter<int>("nSeedsMaxE")),
+
+      etaEdge_(iConfig.getParameter<double>("etaEdge")),
+      mvaCutB_(iConfig.getParameter<double>("mvaCutB")),
+      mvaCutE_(iConfig.getParameter<double>("mvaCutE")),
+
+      minL1Qual_(iConfig.getParameter<int>("minL1Qual")),
+      baseScore_(iConfig.getParameter<double>("baseScore")) {
+  if (!rejectAll_) {
+    mvaEstimator_ = std::make_pair(std::make_unique<SeedMvaEstimator>(mvaFileB_, mvaScaleMeanB_, mvaScaleStdB_),
+                                   std::make_unique<SeedMvaEstimator>(mvaFileE_, mvaScaleMeanE_, mvaScaleStdE_));
+  }
+
+  produces<TrajectorySeedCollection>();
+}
+
+// -- method called on each new Event
+void MuonHLTSeedMVAClassifier::produce(edm::Event& iEvent, edm::EventSetup const& iEventSetup) {
+  auto result = std::make_unique<TrajectorySeedCollection>();
+
+  if (rejectAll_) {
+    iEvent.put(std::move(result));
+    return;
+  }
+
+  if (doSort_ && nSeedsMaxB_ <= 0 && nSeedsMaxE_ <= 0) {
+    iEvent.put(std::move(result));
+    return;
+  }
+
+  if (!doSort_ && mvaCutB_ > 1. && mvaCutE_ > 1.) {
+    iEvent.put(std::move(result));
+    return;
+  }
+
+  edm::ESHandle<TrackerGeometry> trkGeom = iEventSetup.getHandle(trackerGeometryToken_);
```

Comment:
Seems that the `ESHandle` could be avoided
```suggestion
  const TrackerGeometry& trkGeom = iEventSetup.getData(trackerGeometryToken_);
```